File: android/app/src/main/kotlin/com/example/figure_skating_jumps/channels/enums/MethodChannelNames.kt
package com.example.figure_skating_jumps.channels.enums

/**
 * A enum class that contains the method channel names
 *
 * @param channelName The name to identify the method channel
 */
enum class MethodChannelNames(val channelName: String) {
    BluetoothChannel("bluetooth-permission-method-channel"),
    RecordingChannel("recording-method-channel"),
    MeasuringChannel("measuring-method-channel"),
    ConnectionChannel("connection-method-channel"),
    ScanChannel("scan-method-channel")
}

File: ios/Plugins/Channels/Enums/MethodChannelNames.m
#import "MethodChannelNames.h"

NSString *ChannelNames(MethodChannelNames methodChannelName) {
    switch (methodChannelName) {
        case BluetoothChannel:
            return @"bluetooth-permission-method-channel";
        case RecordingChannel:
            return @"recording-method-channel";
        case MeasuringChannel:
            return @"measuring-method-channel";
        case ConnectionChannel:
            return @"connection-method-channel";
        case ScanChannel:
            return @"scan-method-channel";
        default:
            return @"";
    }
}